TemperatureMeter.c

```diff
@@ -0,0 +1,153 @@
+      if (!String_startsWith(buffer, "coretemp") && !String_startsWith(buffer, "cpu_thermal"))
+         continue;
+
+      int m = 0;
+      for (const sensors_feature *feature = sensors_get_features(chip, &m); feature; feature = sensors_get_features(chip, &m)) {
+         if (feature->type == SENSORS_FEATURE_TEMP)
+            count++;
+      }
+
+      result->array = calloc(count, sizeof(int));
+      result->size = count;
+
+      m = 0;
+      for (const sensors_feature *feature = sensors_get_features(chip, &m); feature; feature = sensors_get_features(chip, &m)) {
+         if (feature->type != SENSORS_FEATURE_TEMP)
+            continue;
+
+         const sensors_subfeature *sub_feature = sensors_get_subfeature(chip, feature, type);
+         if (sub_feature) {
+            double local_temp;
+            if (sensors_get_value(chip, sub_feature->number, &local_temp) == 0) {
+               result->array[i] = temp_converter(this, local_temp);
+               ++i;
+            }
+         }
+      }
+   }
+
+   return result;
+}
+
+int TemperatureMeter_attributes[] = {
+   TEMPERATURE_COOL,
+   TEMPERATURE_MEDIUM,
+   TEMPERATURE_HOT,
+};
+
+static void TemperatureMeter_updateValues(Meter* this, char* buffer, int len) {
+   Dynarr* core_temperature = get_cpu_temp(this, SENSORS_SUBFEATURE_TEMP_INPUT);
+   double temperature = core_temperature->array[0];
+
+   // Show only package temperature in bars meter mode
+   if (temperature < 0) {
+      xSnprintf(buffer, len, "N/A");
+      return;
+   }
+   this->total = get_cpu_temp(this, SENSORS_SUBFEATURE_TEMP_MAX)->array[0];
+
+   this->values[0] = temperature;
+   if (this->pl->settings->fahrenheitTemperature)
+      xSnprintf(buffer, len, "%d%sF", (int) temperature, CRT_degreeSign());
+   else
+      xSnprintf(buffer, len, "%d%sC", (int) temperature, CRT_degreeSign());
+   free(core_temperature);
+}
+
+static void TemperatureMeter_display(Object* cast, RichString* out) {
+   Meter* this                = (Meter*)cast;
+   Dynarr* cores_temperatures = get_cpu_temp(this, SENSORS_SUBFEATURE_TEMP_INPUT);
+   int temp_medium            = temp_converter(this, 60);
+   int temp_hot               = temp_converter(this, 70);
+
+   if (cores_temperatures->array[0] < 0 || cores_temperatures->size == 0) {
+      RichString_append(out, CRT_colors[METER_VALUE], "n/a");
+      return;
+   }
+
+   int core_count_begin = cores_temperatures->size == 1 ? 0 : 1;
+
+   for (int core = core_count_begin; core < cores_temperatures->size; ++core) {
+      // convert double to integer
+      int temperature = (int) cores_temperatures->array[core];
+
+      // choose the color for the temperature
+      int tempColor;
+      if      (temperature < temp_medium) tempColor = CRT_colors[TEMPERATURE_COOL];
+      else if (temperature < temp_hot)    tempColor = CRT_colors[TEMPERATURE_MEDIUM];
+      else                                tempColor = CRT_colors[TEMPERATURE_HOT];
```

Comment:
We don't need to save on lines; I'd prefer linebreak (and optionally braces) after the conditions.